src/Billing/Controllers/StripeController.cs

```diff
@@ -471,6 +480,68 @@ public async Task<IActionResult> PostWebhook([FromQuery] string key)
         return new OkResult();
     }
 
+    /// <summary>
+    /// Ensures that the customer associated with the parsed event's data is in the correct region for this server.
+    /// We use the customer instead of the subscription given that all subscriptions have customers, but not all
+    /// customers have subscriptions
+    /// </summary>
+    /// <param name="parsedEvent"></param>
+    /// <returns>true if the customer's region and the server's region match, otherwise false</returns>
+    /// <exception cref="Exception"></exception>
+    private async Task<bool> ValidateCloudRegionAsync(Event parsedEvent)
```

Comment:
What do you think ? If we keep this method light by 
```
var subscriptionEvent = await GetSubscriptionAsync(parsedEvent, true);
return  GetCustomerRegionFromMetadata(subscriptionEvent.Metadata);
```

This way will avoid a lot regression testing on the webhook changes